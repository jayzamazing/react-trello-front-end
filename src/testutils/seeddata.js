'use strict';
import faker from 'faker';
//used to create a single board in json
export const seedBoardsSingle = (title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    title: title,
    _id: _id,
    cardsList: seedCardslistIds()
  };
};
//used to create random boards based on count
export const seedBoardsCount = (count) => {
  if (!count) {
    count = 1;
  }
  let boards = [];
  for (var i = 0; i < count; i++) {
    let _id = faker.random.alphaNumeric(20);
    boards.push({
      _id: _id,
      title: faker.random.words(),
      cardsList: seedCardslistIds(count)
    });
  }
  return boards;
}
//create board items based on input values, after normalization
export const seedBoardsInput = (ids, words, ids2) => {
  let boards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    boards[_id] = {
      _id: _id,
      title: words[i],
      cardslist: ids2[i]
    };
  }
  return boards;
}
//used to create cardslist ids for seedboards
export const seedCardslistIds = (count) => {
  let cardslistIds = [];
  for (var i = 0; i < count; i++) {
    cardslistIds.push(faker.random.alphaNumeric(20));
  }
  return cardslistIds;
}
//used to create a single cardslist in json
export const seedCardslistSingle = (boardId = faker.random.alphaNumeric(20), title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    boardId: boardId,
    title: title,
    _id: _id,
    cards: seedCardsIds()
  };
}
//used to create random cardslist based on count
export const seedCardslistCount = (boardId, count) => {
  if (!count) {
    count = 1;
  }
  let cardslist = [];
  for (var i = 0; i < count; i++) {
    cardslist.push({
      boardId: boardId,
      _id: faker.random.alphaNumeric(20),
      title: faker.random.words(),
      cards: seedCardsIds(count)
    });
  }
  return cardslist;
}
//create cardslist items based on input value, after normalization
export const seedCardslistInput = function(boardId, words, ids) {
  let cardslist = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    cardslist[_id]= {
      boardId: boardId[i],
      _id: _id,
      title: words[i],
      cards: seedCardsIds(ids.length)
    };
  }
  return cardslist;
};
//used to create cardslist ids for seedboards
export const seedCardsIds = (count) => {
  let cardsIds = [];
  for (var i = 0; i < count; i++) {
    cardsIds.push(faker.random.alphaNumeric(20));
  }
  return cardsIds;
}
//used to create a single card in json
export const seedCardsSingle = (cardslistId = faker.random.alphaNumeric(20), text = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    cardslistId: cardslistId,
    text: text,
    _id: _id
  };
}
//used to create random cardslist based on count
export const seedCardsCount = (cardslistId, count) => {
  if (!count) {
    count = 1;
  }
  let cards = [];
  for (var i = 0; i < count; i++) {
    cards.push({
      cardslistId: cardslistId,
      _id: faker.random.alphaNumeric(20),
      text: faker.random.words()
    });
  }
  return cards;
}
//create cards items based on input value, after normalization
export const seedCardsInput = function(cardslistId, words, ids) {
  let cards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = faker.random.alphaNumeric(20);
    cards[_id]= {
      cardslistId: cardslistId[i],
      _id: _id,
      text: faker.random.words()
    };
  }
  return cards;
};
//create a random title
export const createTitle = () => {
return {title: faker.random.words()};
};
