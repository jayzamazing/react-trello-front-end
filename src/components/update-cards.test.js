'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import component, {Cards} from './cards';
import {seedCardslistInput, seedCardsInput} from '../testutils/seeddata';
import UpdateCards from './update-cards';

describe('update-cards', () => {
  let boards = {}, cardslist = {}, cards = {}, params = {}, boardKeys, cardslistKeys, cardsKeys;
  beforeEach(() => {
    let ids = ['46mdujckqs45tkdsh2jdxz', '346rfdghsdhteu021ato', 'kd5o3ve2858qggfqw4rtfda'];
    let ids2 = ['2skdde39hzdo08aqcdgd', '14w43d4opubteu021ato', 'kd5o3ve2858vpjxd7wjp'];
    let words2 = ['XML Health', 'Bedfordshire Illinois Liaison', 'Sports Analyst Cambridgeshire'];
    let ids3 = ['432e39hzdo08aqcdgd', '6532opubteu021ato', '5643e2858vpjxd7wjp'];
    let words3 = ['eat dinner', 'tea tree oil', 'masking tape'];
    cardslist = seedCardslistInput(ids, words2, ids2, ids3);
    cards = seedCardsInput(ids2, words3, ids3);
    cardslistKeys = Object.keys(cardslist);
    cardsKeys = Object.keys(cards);
  });
  afterEach(() => {
    cardslist = {}, cards = {}, cardslistKeys = {}, cardsKeys = {}, boards = {}, params = {}, boardKeys = {};
  });
  it('should render', () => {
    shallow(<UpdateCards _id={cardsKeys[0]} cardslist={cardslist}/>);
  });
  //render and check against previous snapshot
  it('cards snapshot', () => {
    const dispatch = jest.fn();
    const wrapper = shallow(<UpdateCards _id={cardsKeys[0]} cardslist={cardslist}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on spans to see how they modify state
  it('update cards click', () => {
    const wrapper = shallow(<UpdateCards _id={cardsKeys[0]} cardslist={cardslist}/>);
    expect(wrapper.state()).toEqual({ editCardsModalIsOpen: false, modaloredit: true });
    wrapper.find('span').at(0).simulate('click');
    expect(wrapper.state()).toEqual({ editCardsModalIsOpen: true, modaloredit: true });
    wrapper.find('span').at(1).simulate('click');
    expect(wrapper.state()).toEqual({ editCardsModalIsOpen: true, modaloredit: false });
  });
});
