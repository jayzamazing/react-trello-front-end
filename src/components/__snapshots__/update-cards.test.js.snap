// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update-cards cards snapshot 1`] = `
<div
  className="edit-title-or-entire-card"
>
  <span
    onClick={[Function]}
  >
    <Connect(ReduxForm)
      _id="432e39hzdo08aqcdgd"
      disabled="true"
      form="432e39hzdo08aqcdgd-1"
      onBlur={null}
    />
  </span>
  <span
    className="glyphicon glyphicon-pencil edit-cards"
    onClick={[Function]}
  />
  <UpdateCardsModal
    _id="432e39hzdo08aqcdgd"
    cardslist={
      Object {
        "14w43d4opubteu021ato": Object {
          "_id": "14w43d4opubteu021ato",
          "boardId": "346rfdghsdhteu021ato",
          "cards": Array [
            "6532opubteu021ato",
          ],
          "title": "Bedfordshire Illinois Liaison",
        },
        "2skdde39hzdo08aqcdgd": Object {
          "_id": "2skdde39hzdo08aqcdgd",
          "boardId": "46mdujckqs45tkdsh2jdxz",
          "cards": Array [
            "432e39hzdo08aqcdgd",
          ],
          "title": "XML Health",
        },
        "kd5o3ve2858vpjxd7wjp": Object {
          "_id": "kd5o3ve2858vpjxd7wjp",
          "boardId": "kd5o3ve2858qggfqw4rtfda",
          "cards": Array [
            "5643e2858vpjxd7wjp",
          ],
          "title": "Sports Analyst Cambridgeshire",
        },
      }
    }
    closeModal={[Function]}
    isOpen={false}
  />
</div>
`;
